packages/sdks-tests/src/e2e-tests/blocks.spec.ts

```diff
@@ -313,6 +313,41 @@ test.describe('Blocks', () => {
         await expect(textBlock).toHaveText('asfgasgasgasg some test');
       }
     });
+
+    test('video lazy load', async ({ page, sdk }) => {
+      test.skip(checkIsRN(sdk));
+
+      const mockVideoPath = path.join(mockFolderPath, 'video.mp4');
+      const mockVideoBuffer = fs.readFileSync(mockVideoPath);
+      let videoRequestMade = false;
+
+      await page.route('**/*', route => {
+        const request = route.request();
+        if (request.url().includes(VIDEO_CDN_URL)) {
+          videoRequestMade = true;
+
+          return route.fulfill({
+            status: 200,
+            contentType: 'video/mp4',
+            body: mockVideoBuffer,
+          });
+        } else {
+          return route.continue();
+        }
+      });
+
+      await page.goto('/video-lazy-load');
+      const videoLocator = page.locator('video'); 
+
+      await expect(videoLocator).not.toBeInViewport();
+      expect(videoRequestMade).toBeFalsy();
+
+      await videoLocator.scrollIntoViewIfNeeded()
+      await page.waitForTimeout(1000); // allow time for the network request
+
+      await expect(videoLocator).toBeInViewport();
+      expect(videoRequestMade).toBeTruthy();
```

Comment:
we want to avoid using `waitForTimeout` as much as possible. It should never be needed unless under extremely unusual edge-cases.

Adding such timeouts will make our tests unreliable and flaky.

In this case, you want to scroll and then wait for the video request to be made. A better way to do that and not need a timeout is to use `page.waitForResponse()`:

```suggestion
      const requestPromise = page.waitForRequest(request => request.url().includes(VIDEO_CDN_URL))
      await videoLocator.scrollIntoViewIfNeeded()
      await page.waitForTimeout(1000); // allow time for the network request

      await expect(videoLocator).toBeInViewport();
      await requestPromise;
```

The `await requestPromise` will wait until (exactly) one request is made that matches `VIDEO_CDN_URL`, or will throw otherwise. It will take care of waiting the appropriate time so you don't have to